query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -102,37 +102,64 @@ public static Expression createStringArrayLiteralValueExpression(List<String> va
         .build();
   }
 
-  public static boolean isComplexAttributeExpression(Expression expression) {
-    return expression.getValueCase() == ValueCase.ATTRIBUTE_EXPRESSION
-        && expression.getAttributeExpression().hasSubpath();
+  public static boolean isMapAttributeExpression(
+      Expression expression, ViewDefinition viewDefinition) {
+    switch (expression.getValueCase()) {
+      case COLUMNIDENTIFIER:
+        return isMapField(expression.getColumnIdentifier().getColumnName(), viewDefinition);
+      case ATTRIBUTE_EXPRESSION:
+        return expression.getAttributeExpression().hasSubpath()
```

Comment:
A map attribute can have no subpath.

Reply:
Links to contains key for attribute expression. Support not added yet.